File: Python/A/937A-olympiad.py
class Solver:
    def __init__(self):
        self.x = 0
        self.scores = set()

    def input(self):
        self.x = int(input())
        self.scores = set(map(int, input().split()))

    def answer(self):
        return len(self.scores) if 0 not in self.scores else len(self.scores)-1


solution = Solver()
solution.input()
print(solution.answer())

File: C++/A/937A-olympiad.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    int x;
    set<int> scores;

public:
    void input()
    {
        cin >> x;
        for (int i = 0; i < x; i++)
        {
            int a;
            cin >> a;
            if (a > 0)
                scores.insert(a);
        }
    }
    int answer() { return scores.size(); }
};

int main()
{
    Solver solution;
    solution.input();
    cout << solution.answer();
}